systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java

```diff
@@ -131,6 +135,36 @@ void assertThatRollingUpdatedFinished(String rolledComponent, String stableCompo
         assertThat("", statusCount.get("Running"), is(Integer.toUnsignedLong(podStatuses.size())));
     }
 
+    @Description("Test for checking that overriding of bootstrap server, triggers the rolling update.")
+    @Test
+    void testTriggerRollingUpdateAfterOverrideBootstrap() {
+        String bootstrapDns = "kafka-test.XXXX.azure.XXXX.net";
+
+        testMethodResources().kafkaEphemeral(CLUSTER_NAME, 3, 1).done();
+
+        Map<String, String> kafkaPods =  StUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            kafka.getSpec().getKafka().getListeners().setExternal(
+                new KafkaListenerExternalLoadBalancerBuilder()
+                    .withNewOverrides()
+                        .withNewBootstrap()
+                            .withAddress(bootstrapDns)
+                        .endBootstrap()
+                    .endOverrides()
+                    .build());
+        });
+
+        StUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
+        StUtils.waitUntilKafkaCRIsReady(CLUSTER_NAME);
+
+        String bootstrapAddressDns = ((KafkaListenerExternalLoadBalancer) Crds.kafkaOperation(kubeClient().getClient())
```

Comment:
This really just verifies that you updated the CR. Would it be worth to also check that the CO does what it is supposed to and the new bootstrap name is in the alternate DNS subjects in the certificates?

Reply:
Did it already )